io/eolearn/io/processing_api.py

```diff
@@ -61,16 +64,17 @@ def __init__(self, data_source, size=None, resolution=None, cache_folder=None, c
         :type config: SHConfig or None
         :param max_threads: Maximum threads to be used when downloading data.
         :type max_threads: int
+        :param data_source: A deprecated alternative to data_collection
+        :type data_source: DataCollection
         """
-
         if (size is None) == (resolution is None):
             raise ValueError("Exactly one of the parameters 'size' and 'resolution' should be given.")
 
         self.size = size
         self.resolution = resolution
         self.config = config or SHConfig()
         self.max_threads = max_threads
-        self.data_source = data_source
+        self.data_collection = DataCollection(handle_deprecated_data_source(data_collection, data_source))
```

Comment:
There is a bit of mismatch how `handle_deprecated...` is used throughout the code. Sometimes it is passed to the  `DataCollection` constructor (as in this case), while some other times it is not. 

I suggest this is consolidated and handled in the utility function (so that it returns `DataCollection`).

Reply:
I have made a small update regarding this in `cloud_mask.py` so that we now have `DataCollection` casting everywhere. In the future, `handle_deprecated_data_source` will be removed and `DataCollection` casting will remain. Because of that I wouldn't put `DataCollection` casting into `handle_deprecated_data_source`.